woocommerce/woocommerce-android
# Perform mark complete swipe action

Closes: #7035
Closes: #7036

‼️ PR depends on [this fix](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2475) on FluxC

### Description
This PR updates the order's status when you swipe the order to the left. With these changes, the app retrieves the order id, position, and current status to the ViewModel once the Order item is swiped. With this information, the ViewModel optimistically changes the order status locally to _completed_, shows a message indicating the change to the merchant (with an undo action), and performs a request on the API to update the status. 
If the request fails, it displays a retry SnackBar to the merchant so he can retry the action. 
If the merchant presses the undo action, the order status changes to the previous status.

The Add (+) FAB button should be a direct child of a CoordinatorLayout to let the layout handles the coordinated animation when the SnackBar is shown, but I will create a different issue for this fix because is out of the scope of this PR.

### Testing instructions
TC1 
1. Open orders
2. Swipe a non-completed order
3. Check that the order changes status to completed 
4. Check that the app shows "Order #OrderId marked as completed"
5. Press undo in the snack bar
6. Check that the order updates the status to the status before

TC2 
1. Open orders
2. Turn on airplane mode
3. Swipe a non-completed order
4. Check that the order maintains the current status
5. Check that an error message is displayed

TC3
Run [OrderListViewModelTest](https://github.com/woocommerce/woocommerce-android/blob/4ab2eca382af369b0af915b82160496c8a278915/WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderListViewModelTest.kt)


### Images/gif

#### Swipe-To Complete success

https://user-images.githubusercontent.com/18119390/182223840-8e497973-358b-4daf-b2a0-e0a0746c8256.mp4

#### Swipe-To Complete fail

https://user-images.githubusercontent.com/18119390/182223885-8f9dd4c5-c3ca-4913-a536-2a93e222b71c.mp4
